File: Fronted/gestion-tareas/src/app/modules/tareas/domain/ITareaRepository.ts
import { TareaModel } from "src/app/core/models/TareaModel";
import { ApiResponse } from "../application/dtos/response/api-response.model";
import { GestionTareasResponse } from "../application/dtos/response/gestion-tareas-response";
import { TareaRequest } from "../application/dtos/request/tarea-request.model";

export interface ITareaRepositorio {
    gestionTareas(): Promise<ApiResponse<GestionTareasResponse[]>>;
    getAllHistorico(): Promise<ApiResponse<GestionTareasResponse[]>>;
    inactivateById(id: number): Promise<ApiResponse<string>>;
    activarById(tareaRequest: TareaRequest): Promise<ApiResponse<string>>;
    insert(tareaRequest: TareaRequest): Promise<ApiResponse<string>>;
  }

File: Backend/gestion-tareas/gestion-tareas/Repository/Interfaces/ITareaRepository.cs
using gestion_tareas.DTOs.Request;
using gestion_tareas.DTOs.Response;
using gestion_tareas.Models;

namespace gestion_tareas.Repository.Interfaces
{
    public interface ITareaRepository
    {
        List<GestionTareasResponse> getAllTareas();
        public List<GestionTareasResponse> getAllHistorico();

        int InsertTask(Tarea tarea);
        int UpdateTask(Tarea tarea);
        int InactivarTask(int id);
        int ActivarTask(int idTarea);
    }
}
